File: Singleton/C++/Singleton.cpp
#include "Singleton.h"
#include <iostream>
#include <string>

Singleton::Singleton()
    : totalCounts(0)
{

}

std::ostream& operator<< (std::ostream& os, const Singleton& s)
{
    os << "Singleton instance to string"; 
    return os;
}

void Singleton::echo(const std::string& str) const
{
    std::cout << str << std::endl;
}

void Singleton::incrementCount()
{
    totalCounts++;
}

int Singleton::getCount()
{
    return totalCounts;
}

File: Singleton/C#/Singleton.cs
public class Singleton
{
    private static readonly Singleton instance = new Singleton();

    static Singleton()
    {

    }

    private Singleton()
    {

    }

    public static Singleton Instance
    {
        get { return instance; }
    }

    public void echo(string line)
    {
        System.Console.WriteLine(line);
    }

    public int add(int a, int b)
    {
        return a+b;
    }
}